joblib/test/test_parallel.py

```diff
@@ -940,3 +940,32 @@ def __reduce__(self):
     with warns(DeprecationWarning):
         with raises(ValueError):
             delayed(UnpicklableCallable(), check_pickle=True)
+
+
+def _backend_reset_func(x):
+    for x in range(1000):
+        x ** x
+    return x ** 2
+
+
+def test_backend_reset():
+    """Test that a backend is correctly reset to its initial state."""
+
+    # Use default backend
+    start_time = time.time()
+    Parallel(n_jobs=4)(delayed(_backend_reset_func)(i) for i in range(100))
+    ref_time = time.time() - start_time
+
+    p = Parallel(n_jobs=4)
+    start_time = time.time()
+    p(delayed(_backend_reset_func)(i) for i in range(100))
+    first_time = time.time() - start_time
+    assert p._backend.__dict__ == {}
+
+    start_time = time.time()
+    p(delayed(_backend_reset_func)(i) for i in range(100))
+    second_time = time.time() - start_time
+    assert p._backend.__dict__ == {}
+
+    assert first_time <= ref_time
+    assert second_time <= ref_time
```

Comment:
I have mixed feelings about tests that rely on timing. I am pretty sure they will fail randomly on travis where travis workers tend to be shared by many projects.

Reply:
Yep agreed.